src/main/java/org/opentripplanner/index/GeocoderResource.java

```diff
@@ -36,9 +37,24 @@ public GeocoderResource (@Context OTPServer otpServer, @PathParam("routerId") St
         }
     }
 
+    /**
+     * Geocode using data using the OTP graph for stops, clusters and street names
+     *
+     * @param query The query string we want to geocode
+     * @param autoComplete Whether we should use the query string to do a prefix match
+     * @param showStops Search for stops, either for name or stop code
+     * @param showClusters Search for clusters by their name
+     * @param showCorners Search for street corners using at least one of the street names
+     * @return list of results in in the format expected by GeocoderBuiltin.js in the OTP Leaflet client
+     */
     @GET
-    public Response textSearch (@QueryParam("query") String query) {
-        return Response.status(Response.Status.OK).entity(index.query(query)).build();
+    public Response textSearch (@QueryParam("query") String query,
+                                @QueryParam("autoComplete") @DefaultValue("false") boolean autoComplete,
+                                @QueryParam("showStops") @DefaultValue("true") boolean showStops,
```

Comment:
Minor point: In the interest of brevity we should probably just reduce these parameter names to stops, clusters, and corners. Besides being shorter, setting them to true does not exactly "show" the elements, it includes them in the response.